destination/destination.go

```diff
@@ -107,26 +109,23 @@ func (d *Destination) Open(ctx context.Context) error {
 		return nil
 	}
 
-	err := backoff.Retry(func() error {
-		streamData, err := d.client.DescribeStream(ctx, &kinesis.DescribeStreamInput{
-			StreamName: &d.config.StreamName,
-		})
+	if d.config.CreateIfNotExists {
+		exists, err := d.doesStreamExist(ctx, d.config.StreamName)
 		if err != nil {
-			return fmt.Errorf("failed to describe stream: %w", err)
+			return fmt.Errorf("failed to check if stream exists: %w", err)
+		} else if exists {
+			return nil
 		}
 
-		switch status := streamData.StreamDescription.StreamStatus; status {
-		case types.StreamStatusCreating, types.StreamStatusUpdating, types.StreamStatusDeleting:
-		case types.StreamStatusActive:
-			sdk.Logger(ctx).Info().Msg("destination ready to be written to")
-			return nil
+		if err := d.createStream(ctx, d.config.StreamName); err != nil {
+			return fmt.Errorf("failed to create stream %s: %w", d.config.StreamName, err)
```

Comment:
If `createStream` errors, error would be nested ("failed to create stream %s: failed to create stream %s"). I think this `Open` method should wrap errors in a way that we could easily determine where the error is actually coming from. Probably the same in line [#115](https://github.com/conduitio-labs/conduit-connector-kinesis/pull/69/files#diff-7fd240e53ce4c101a889038fcb325fdd36d10e18e771eaa78fdbbf8ea8c700a3R115.)

Reply:
Good catch @raulb

Reply:
I don't really know what error context to give though. If returning the error as is, the errwrap linter doesn't complain. Should we just leave it as is?